sos/utilities.py

```diff
@@ -471,8 +471,6 @@ def get_contents(self):
         """Returns the contents of the deque as a string"""
         # block until command completes or timesout (separate from the plugin
         # hitting a timeout)
-        while self.running:
-            time.sleep(0.01)
```

Comment:
This is what blocks until the command completes or is killed by timeout. This should stay.

Reply:
If the 'fwdownloader' command's child-process is deadlock,  self.running is always True, this would cause deadlock in this get_contents function.

In the exception handling,  this get_contents function is invoked. 
return {'status': p.poll(), 'output': reader.get_contents(),'truncated': reader.is_full}

We have two option to prevent deadlock
1. Delete running validation in get_contents function
2. Delete reader.get_contents() in Exception handling 
OR
We could create a new Exception (let's say ChildDeadlockException )
If exception = ChildDeadlockException , output='', else output=reader.get_contents()